compiler/src/dotty/tools/dotc/typer/Inferencing.scala

```diff
@@ -81,6 +81,19 @@ object Inferencing {
     if (depVars.nonEmpty) instantiateSelected(tp, depVars.toList)
   }
 
+  /** If `tp` is type variable with a lower bound in the current constraint,
+   *  instantiate it from below.
+   */
+  def couldInstantiateTypeVar(tp: Type)(using Context): Boolean = tp.dealias match
```

Comment:
I don't think that's quite right either since given `?F[?A]` I need to instantiate at least `?F`, and then if it's a type lambda I may or may not need to instantaite `?A` too. Also since this is a selection prefix, we want to avoid `Nothing` if possible, in fact in all situation where we're trying to avoid `Nothing`, we only want to avoid it at the top-level. This is what I came up last time I was looking at this: https://github.com/dotty-staging/dotty/blob/63c7bad7f01c5fccf4620376aa7c8f70bc90b66d/compiler/src/dotty/tools/dotc/typer/Inferencing.scala#L49-L83
but I was a bit concerned that this duplicated code from IsFullyDefinedAccumulator.

Reply:
Right. In this particular case we need to instantiate the `?F`. The `?A` can come later when we recursively call `typedSelect` on the new call. I have taken your changes and adapted them to `couldInstantiateTypeVar`. You don't need to check for `LazyRef`  in either code, since that is already eliminated by `dealias`. On the other hand, one should look inside AndTypes, OrTypes, and AnnotatedTypes.